frontend/src/utils/ews-utils.ts

```diff
@@ -57,16 +57,19 @@ type EWSTriggerLevels = {
 export const createEWSDatesArray = (testEndDate?: number): number[] => {
   const datesArray = [];
 
-  const endDate =
-    testEndDate || moment(moment.utc().format('YYYY-MM-DD')).valueOf();
+  const now = new Date();
 
-  const tempDate = moment.utc('2021-01-01');
+  const endDate = testEndDate || now.setUTCHours(0, 0, 0, 0);
+
+  let tempDate = new Date('2021-01-01');
 
   while (tempDate.valueOf() <= endDate) {
+    const clone = new Date(tempDate.getTime());
     // eslint-disable-next-line fp/no-mutating-methods
-    datesArray.push(tempDate.clone().set({ hour: 12, minute: 0 }).valueOf());
+    datesArray.push(clone.setHours(12));
```

Comment:
we also need to set the minutes to 0 for 30min timezones
it might be worth creating a utility `setToNoon` to reuse across the codebase.